### PR TITLE
feat(auth): add verifyTOTPSetup API

### PR DESCRIPTION
**Pull request by israx**
#### Description of changes

`verifyTOTPSetup` API is used to finish the TOTP process. It verifies the OTP code returned by the authenticator app that was associated by first calling the `setUpTOTP` [API](https://github.com/aws-amplify/amplify-js/pull/11635).

**TODO list:**

- get the accessToken from the AuthToken provider.

#### Description of how you validated changes

In my local machine, I used the `CognitoUserPoolTokenManager.setTokens` method to set tokens on a successful call of `signIn`. Then I used the `CognitoUserPoolTokenManager.loadTokens` method inside `verifyTOTPSetup` API to grab the `accessToken` and pass it to the client.

Then I created a sample App to test the functionality, and was able to verify TOTP. 

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/providers/cognito/apis/verifyTOTPSetup.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { CogntioVerifyTOTPSetupOptions } from '../types/options';
+import { verifySoftwareTokenClient } from '../utils/clients/VerifySoftwareTokenClient';
+import { VerifySoftwareTokenException } from '../types/errors';
+
+/**
+ * Verifies a otp code retrieved from an associated authenticated app.
+ *
+ * @param verifyTOTPSetupRequest - The VerifyTOTPSetupRequest
+ *
+ * @throws  -{@link VerifySoftwareTokenException }:
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/apis/verifyTOTPSetup.ts:16]:
```suggestion
 * @throws {@link VerifySoftwareTokenException}
```

File: packages/auth/src/providers/cognito/apis/verifyTOTPSetup.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * Thrown due to an invalid MFA token.
+ *
+ * @throws  -{@link AuthValidationErrorCode }:
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/apis/verifyTOTPSetup.ts:19]:
```suggestion
 * @throws {@link AuthValidationErrorCode}
```

**Review by jimblanc — DISMISSED** [review]
Looks good! Few doc nits below (feel free to ignore at your discretion).

File: packages/auth/src/providers/cognito/apis/verifyTOTPSetup.ts (hunk 1 above)
Comment (israx) [reply packages/auth/src/providers/cognito/apis/verifyTOTPSetup.ts:16]: type doc weird intellisense as discussed on a previous PR

File: packages/auth/src/providers/cognito/apis/verifyTOTPSetup.ts (hunk 2 above)
Comment (israx) [reply packages/auth/src/providers/cognito/apis/verifyTOTPSetup.ts:19]: type doc weird intellisense as discussed on a previous PR

**Review by jimblanc — APPROVED** [review]

**Review by elorzafe — DISMISSED** [review]
great work @israx

**Review by elorzafe — APPROVED** [review]
Great work @israx 🎖️

**Review by jimblanc — APPROVED** [review]

**Pull request merged**